3567/0/about_scoring_project.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+require File.expand_path(File.dirname(__FILE__) + '/neo')
+
+# Greed is a dice game where you roll up to five dice to accumulate
+# points.  The following "score" function will be used to calculate the
+# score of a single roll of the dice.
+#
+# A greed roll is scored as follows:
+#
+# * A set of three ones is 1000 points
+#
+# * A set of three numbers (other than ones) is worth 100 times the
+#   number. (e.g. three fives is 500 points).
+#
+# * A one (that is not part of a set of three) is worth 100 points.
+#
+# * A five (that is not part of a set of three) is worth 50 points.
+#
+# * Everything else is worth 0 points.
+#
+#
+# Examples:
+#
+# score([1,1,1,5,1]) => 1150 points
+# score([2,3,4,6,2]) => 0 points
+# score([3,4,5,3,3]) => 350 points
+# score([1,5,1,2,4]) => 250 points
+#
+# More scoring examples are given in the tests below:
+#
+# Your goal is to write the score method.
+
+def check_set_of_three_fives(array)
+  if array[4] >= 3
+    array[4] -= 3
+    return 500
+  end
+  0
+end
+
+def check_set_of_three_ones(array)
+  if array[0] >= 3
+    array[0] -= 3
+    return 1000
+  end
+  0
+end
+
+def check_set_of_three(array, iteration)
+  return 100 * (iteration + 1) if array[iteration] == 3
+
+  0
+end
+
+def check_everything(array)
+  result = 0
+  (0..5).each do |iteration|
+    result += check_set_of_three(array, iteration)
+    if (array[iteration] < 3) && array[iteration].positive?
+      result += 50 * array[iteration] if iteration == 4
+      result += 100 * array[iteration] if iteration.zero?
+    end
+  end
+  result
+end
+
+def score(dice)
+  array = []
+  result = 0
+  (1..6).each do |iteration|
+    array.append(dice.select { |item| item == iteration }.count)
```

Comment:
O_o omg 🤦‍♂ `dice.count(number)` ?
Remove `result = 0`
Use `reduce, inject, each_with_object` instead of `each`